platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -316,6 +316,11 @@ public static class ClassVersion {
 
     private VirtualMapStatistics statistics;
 
+    /**
+     * This reference is used to assert that there is only one thread modifying the VM at a time.
+     */
+    private AtomicReference<Thread> currentModifyingThreadRef;
```

Comment:
Please, add a comment to JavaDoc that this field is only used in assertions. If assertions are disabled, the field is always null.

Reply:
Done